### PR TITLE
Fixed logic for getting files and folder by path

### PR DESCRIPTION
**Pull request by PauliusGrigaliunas**

### CODE REVIEW

File: clients/projectshare/src/ProjectShareClient.ts [hunk 1]
```diff
@@ -518,6 +550,32 @@ export class ProjectShareClient extends WsgClient {
     return this.postInstance<ProjectShareFile>(requestContext, ProjectShareFile, relativeUrl, updateInstance);
   }
 
+  /**
+   * Delete folder permanently or send folder to recycle bin.
+   * @param requestContext Client request context that includes the authorization information to access Project Share.
+   * @param contextId Context Id (e.g., projectId or assetId)
+   * @param folderId Id of the folder where we need to upload that file.
```
Comment (tomaspakala) [inline clients/projectshare/src/ProjectShareClient.ts:557]: Should be:
* @param folderId of the folder you want to delete.

File: clients/projectshare/src/ProjectShareClient.ts [hunk 2]
```diff
@@ -518,6 +550,32 @@ export class ProjectShareClient extends WsgClient {
...
+        updateInstance.changeState = "modified";
+        const projectShareRequestOptions = {
+          CustomOptions: { // eslint-disable-line @typescript-eslint/naming-convention
+            IsDeleted: true, // eslint-disable-line @typescript-eslint/naming-convention
+          },
+        };
+        return this.postInstance<ProjectShareFolder>(requestContext, ProjectShareFolder, url, updateInstance, projectShareRequestOptions);
+    }
+  }
+
```
Comment (tomaspakala) [inline clients/projectshare/src/ProjectShareClient.ts:577]: Single new line

Comment (tomaspakala) [inline clients/projectshare/src/ProjectShareClient.ts:577]: Single new line

File: clients/projectshare/src/ProjectShareClient.ts (hunk 1 above)
Comment (tomaspakala) [inline clients/projectshare/src/ProjectShareClient.ts:557]: Should be:
*@param folderId of the folder you want to delete.

**Review by tomaspakala — COMMENTED** [review]
Mainly some naming and code documentation should be updated a bit

**Comment by calebmshafer** [comment]
@PauliusGrigaliunas can you sign the CLA?  I'll kick off a new build as well.

**Comment by calebmshafer** [comment]
@PauliusGrigaliunas it looks like the build is failing with the changelog files. Can you run `rush change`?

**Review by calebmshafer — CHANGES_REQUESTED** [review]
it looks like the build is failing with the changelog files. Can you run `rush change` to add one?

**Review by calebmshafer — CHANGES_REQUESTED** [review]
@PauliusGrigaliunas it looks like there are a few lint failures that still need to be cleaned up. Mind taking a look at them?  https://dev.azure.com/imodeljs/imodeljs/_build/results?buildId=2390&view=logs&j=49d0b678-52ae-56ca-d936-f4e8771f3f89

**Comment by pgrigaliunas** [comment]
> @PauliusGrigaliunas it looks like there are a few lint failures that still need to be cleaned up. Mind taking a look at them? https://dev.azure.com/imodeljs/imodeljs/_build/results?buildId=2390&view=logs&j=49d0b678-52ae-56ca-d936-f4e8771f3f89

solved

**Comment by calebmshafer** [comment]
@PauliusGrigaliunas going to kick off a set of builds for this now. However I noticed that #340 had a number of similar changes?  Are both of these PRs still necessary and if so is there a specific order to them?

**Comment by tomaspakala** [comment]
@calebmshafer #340 is based on this branch. There is no order but Paulius was working on improving integration tests in another branch while waiting for this branch to be merger.

**Review by calebmshafer — APPROVED** [review]

**Review by calebmshafer — CHANGES_REQUESTED** [review]
@PauliusGrigaliunas can you run `rush extract-api` after rebuilding?  The CI is currently failing due to that.

**Comment by calebmshafer** [comment]
Closing in favor of #340.

**Pull request closed without merging**